src/OrbitGl/CaptureWindowTest.cpp

```diff
@@ -79,4 +81,193 @@ TEST_F(UnitTestCaptureWindow, SlidersRespondToMouseLeave) {
+  [[nodiscard]] orbit_capture_client::CaptureClient::State GetCaptureState() const {
+    return orbit_capture_client::CaptureClient::State::kStopped;
+  }
+
+  [[nodiscard]] bool IsCapturing() const { return false; }
+
+  void StartCapture(){};
+  void StopCapture(){};
+  void AbortCapture(){};
+  void ToggleCapture(){};
+};
+
+class NavigationTestCaptureWindow : public CaptureWindow, public testing::Test {
+ public:
+  explicit NavigationTestCaptureWindow() : CaptureWindow(nullptr) {
+    capture_client_app_ = &capture_client_app_fake_;
+
+    Resize(600, 100);
+
+    capture_data_ = TrackTestData::GenerateTestCaptureData();
+    time_graph_ = std::make_unique<TimeGraph>(this, nullptr, &viewport_, capture_data_.get(),
+                                              &picking_manager_);
+
+    CreateSchedulerTimers();
+
+    // Prepare the TimeGraph to be fully zoomed out
+    UpdateChildrenPosAndSize();
+    time_graph_->ZoomAll();
+    time_graph_->UpdateLayout();
+    UpdateChildrenPosAndSize();
+
+    // Make sure our expectations about the height of the timeline are correct,
+    // otherwise zooming may not work as expected - if this fails, the overall
+    // viewport height should be changed and the test adjusted accordingly
+    assert(time_graph_->GetTimelineUi()->GetHeight() < 95);
+    assert(time_graph_->GetTimelineUi()->GetPos()[0] == 0);
+  }
+
+ protected:
+  void ExpectInitialState() {
+    UpdateChildrenPosAndSize();
+
+    EXPECT_LT(vertical_slider_->GetLengthRatio(), 1.0);
+    EXPECT_EQ(vertical_slider_->GetPosRatio(), 0.0);
+
+    EXPECT_EQ(slider_->GetLengthRatio(), 1.0);
+    EXPECT_EQ(slider_->GetPosRatio(), 0.0);
+    EXPECT_EQ(time_graph_->GetMaxTimeUs() * 1000, time_graph_->GetCaptureMax());
+    EXPECT_EQ(time_graph_->GetMinTimeUs() * 1000, 0);
+  }
+
+  enum struct PosWithinCapture { kLeft, kRight, kInBetween };
+  void ExpectIsHorizontallyZoomedIn(PosWithinCapture pos) {
+    UpdateChildrenPosAndSize();
+
+    EXPECT_LT(slider_->GetLengthRatio(), 1.0);
+    switch (pos) {
+      case PosWithinCapture::kLeft:
+        EXPECT_EQ(slider_->GetPosRatio(), 0.0);
+        EXPECT_LT(time_graph_->GetMaxTimeUs() * 1000, time_graph_->GetCaptureMax());
+        EXPECT_EQ(time_graph_->GetMinTimeUs() * 1000, 0);
+        break;
+      case PosWithinCapture::kRight:
+        EXPECT_GT(slider_->GetPosRatio(), 0.0);
+        EXPECT_EQ(time_graph_->GetMaxTimeUs() * 1000, time_graph_->GetCaptureMax());
+        EXPECT_GT(time_graph_->GetMinTimeUs() * 1000, 0);
+        break;
+      case PosWithinCapture::kInBetween:
+        EXPECT_GT(slider_->GetPosRatio(), 0.0);
+        EXPECT_LT(time_graph_->GetMaxTimeUs() * 1000, time_graph_->GetCaptureMax());
+        EXPECT_GT(time_graph_->GetMinTimeUs() * 1000, 0);
+        break;
+    }
+  }
+
+ private:
+  CaptureClientAppInterfaceFake capture_client_app_fake_;
+  std::unique_ptr<orbit_client_data::CaptureData> capture_data_;
+
+  void CreateSchedulerTimers() {
```

Comment:
I feel that those timers should be more close to `TrackTestData`, as there is in general a semantic relation ship between our timers and the capture data. If we would add additional tests in the future that could lead to inconsistencies.

Reply:
True, I've moved this. I'm a bit worried that the current track test data is very specific to a few tests, I basically just added what I needed when I needed it, and we might end up breaking a lot of tests when we expand this. But we probably just have to start some place.

Reply:
>I'm a bit worried that the current track test data is very specific to a few tests

Yeah, I can see that. Maybe we can get some very basic test data and then in the tests extend them in a centralized way. As said, my main concern was that capture data (including timers) is something that need to be consistent. So it should be clear what is going on there.